File: TrainingCshar/Interfaces/Collections/IPila.cs
using System.Collections.Generic;

namespace TrainingCshar.Collections
{
    internal interface IPila
    {
        void Agregar(PersonaPila persona);

        void Agregar(Stack<PersonaPila> stack);

        string Cantidad();

        Stack<PersonaPila> Clonar();

        void Eliminar(bool todos);

        List<string> Imprimir();
    }
}

File: TrainingCshar/Interfaces/Collections/IPila.vb
Imports System.Collections.Generic

Namespace TrainingCshar.Collections
    Friend Interface IPila
        Sub Agregar(ByVal persona As Persona)
        Sub Agregar(ByVal stack As Stack(Of Persona))
        Function Cantidad() As String
        Function Clonar() As Stack(Of Persona)
        Sub Eliminar(ByVal todos As Boolean)
        Function Imprimir() As List(Of String)
    End Interface
End Namespace
